siad/daemon.go

```diff
@@ -32,82 +35,141 @@ func processNetAddr(addr string) string {
 	return addr
 }
 
+// processModules makes the modules string lowercase to make checking if a
+// module in the string easier, and returns an error if the string contains an
+// invalid module character.
+func processModules(modules string) (string, error) {
+	modules = strings.ToLower(modules)
+	validModules := "cghmrtwe"
+	invalidModules := modules
+	for _, m := range validModules {
+		invalidModules = strings.Replace(invalidModules, string(m), "", 1)
+	}
+	if len(invalidModules) > 0 {
+		return "", errors.New("Unable to parse --modules flag, unrecognized modules: " + invalidModules)
```

Comment:
this error will be returned if you supply duplicate modules. Kind of misleading the modules aren't unrecognized, just duplicates.